packages/block-editor/src/components/rich-text/use-delete.js

```diff
@@ -43,7 +43,7 @@ export function useDelete( props ) {
 				// an intentional user interaction distinguishing between Backspace and
 				// Delete to remove the empty field, but also to avoid merge & remove
 				// causing destruction of two fields (merge, then removed merged).
-				if ( onRemove && isEmpty( value ) && isReverse ) {
+				else if ( onRemove && isEmpty( value ) && isReverse ) {
```

Comment:
With this change in practice, this condition `else if ( onRemove && isEmpty( value ) && isReverse ) {`is never executed right ? We pass an onMerge on the block list so `if ( onMerge ) {` will always be truth, if that's the case would it make sense to remove the condition?

Reply:
Not all blocks may pass an onMerge function to RichText, they may only pass onRemove.